src/pudl_archiver/depositors/zenodo.py

```diff
@@ -0,0 +1,319 @@
+
+        async def requester(
+            method: str, url: str, log_label: str, parse_json: bool = True, **kwargs
+        ) -> dict | aiohttp.ClientResponse:
+            """Make requests to Zenodo.
+
+            Args:
+                method: HTTP method - "GET", "POST", etc.
+                url: the URL you are going to hit.
+                log_label: a string describing what this request does, for
+                    logging purposes.
+                parse_json: whether or not to always parse the response as a
+                    JSON object. Default to True.
+
+            Returns:
+                Either the parsed JSON or the raw aiohttp.ClientResponse object.
+            """
+            logger.info(f"{method} {url} - {log_label}")
+            async with session.request(method, url, **kwargs) as response:
+                if response.status >= 400:
+                    raise ZenodoClientException(
+                        {"response": response, "json": await response.json()}
+                    )
+                if parse_json:
+                    return await response.json()
+                return response
+
+        return requester
+
+    async def create_deposition(self, metadata: DepositionMetadata) -> Deposition:
+        """Create a whole new deposition.
+
+        Args:
+            metadata: a metadata, to make a deposition with.
+
+        Returns:
+            The brand new deposition - unpublished and with no files in it.
+        """
+        url = f"{self.api_root}/deposit/depositions"
+        headers = {
+            "Content-Type": "application/json",
+        } | self.auth_write
+
+        payload = {
+            "metadata": metadata.dict(
+                by_alias=True,
+                exclude={"publication_date", "doi", "prereserve_doi"},
+            )
+        }
+
+        response = await self.request(
+            "POST", url, "Create new deposition", json=payload, headers=headers
+        )
+        # Ignore content type
+        return Deposition(**response)
+
+    async def get_deposition(self, concept_doi: str) -> Deposition:
+        """Get the latest deposition associated with a concept DOI.
+
+        Sometimes the deposition information that comes back from the concept
+        DOI query is incomplete, so we use the record ID that is returned from
+        that to make another request for the "full" data.
+
+        Args:
+            concept_doi: the DOI for the concept - gets the latest associated
+                DOI.
+
+        Returns:
+            The latest deposition associated with the concept DOI.
+        """
+        url = f"{self.api_root}/deposit/depositions"
+        params = {"q": f'conceptdoi:"{concept_doi}"'}
+
+        response = await self.request(
+            "GET",
+            url,
+            f"Query depositions for {concept_doi}",
+            params=params,
+            headers=self.auth_write,
+        )
+        if len(response) > 1 and not self.sandbox:
```

Comment:
Since the concept DOI is defined as "always pointing at the latest deposition" we can just always get the most recent one. We should still log if the response is plural though.